src/cli/commands/publish.js

```diff
@@ -33,6 +35,19 @@ async function publish(
   if (access && access !== 'public' && access !== 'restricted') {
     throw new MessageError(config.reporter.lang('invalidAccess'));
   }
+  
+  commonDirectories.map((dir) => {
+    if (pkg[dir]) {
+      for (const file in pkg[dir]) {
+        fs.access(config.cwd + '/' + pkg[dir][file].replace('./', '')).then((res) => {
+          // valid files
+        }).catch((err) => {
+          config.reporter.error(`file not found: ${err.path}`);
+          process.exit(1);
+        });
```

Comment:
> **better use async/awaut than promisses**  

Good catch. Will change.

> **process.exit does not seem like the right way to do, a command should not terminate process, we have a top level function for that in cli/index.js**

Are you referring to this function inside of [cli/index.js](https://github.com/yarnpkg/yarn/blob/099e76dc075763427605a6a63906127402370bb2/src/cli/index.js#L315) ? I was roughly following [this pattern](https://github.com/yarnpkg/yarn/blob/099e76dc075763427605a6a63906127402370bb2/src/cli/index.js#L168), and verbose logging for a potentially simple user error (not compiled, typo, etc.) didn't seem necessary .

> **I don't understand the logic here: it checks that user has access to folders bin/main/modules/files folders? What does it verify?**

The feature mentioned in #1930 cites common files that exist in directories that are going to be published. 

> A couple times I published packages without the required files or with the wrong paths in these properties:

> bin
main
module
files

To attempt to solve the user error, I have this logic:  

1.) Keep a list of ["common" directories](https://github.com/olingern/yarn/blob/6a961258fb087a972079ae20927e2e7597f691d2/src/util/common-dir.js).
2.) Compare this list with the user's package.json file  
  *  If a match is found, attempt to access it:  
    ``` fs.access(config.cwd + '/' + pkg[dir][file].replace('./', '')) ```
  * If we come across an entry in package.json that cannot be accessed, then abort publishing and notify the user.  

    ```     
          }).catch((err) => {
            config.reporter.error(`file not found: ${err.path}`);
            process.exit(1);
          }); 
    ```